--- a/Source/extensions/privilegedrequest/include/privilegedrequest/PrivilegedRequest.h
+++ b/Source/extensions/privilegedrequest/include/privilegedrequest/PrivilegedRequest.h
@@ -335,9 +335,10 @@ PUSH_WARNING(DISABLE_WARNING_UNUSED_RESULT)
 
                 ASSERT(strlen(binder) < MaxFilePathLength);
-PUSH_WARNING(DISABLE_WARNING_UNUSED_RESULT)
-                ::mkstemp(binder);
-POP_WARNING()
-                return (string(binder));
+                int fd = ::mkstemp(binder);
+                if (fd >= 0) {
+                    ::close(fd);
+                }
+                return (string(binder));
             }
