diff --git a/Tests/unit/CMakeLists.txt b/Tests/unit/CMakeLists.txt
index d13f472f..beb2d6ec 100644
--- a/Tests/unit/CMakeLists.txt
+++ b/Tests/unit/CMakeLists.txt
@@ -18,8 +18,10 @@ find_package(GTest REQUIRED)
 find_package(GTest REQUIRED)
 
 # TODO: coverage causes attempt to write on invalid location, -keep-static-functions is not recognized by buildroot compiler
-add_compile_options(--coverage -fno-inline -fno-inline-small-functions -fno-default-inline)
-add_link_options(-fprofile-arcs)
+if (ENABLE_CODE_COVERAGE)
+    add_compile_options(--coverage -fno-inline -fno-inline-small-functions -fno-default-inline)
+    add_link_options(-fprofile-arcs)
+endif()
 add_compile_options(-pthread)
 add_link_options(-pthread)
